lib/full-site-editing/class-gutenberg-rest-templates-controller.php

```diff
@@ -340,18 +340,10 @@ protected function prepare_item_for_database( $request ) {
 		if ( null === $template ) {
 			$changes->post_type   = $this->post_type;
 			$changes->post_status = 'publish';
-			$changes->tax_input   = array(
-				'wp_theme' => isset( $request['theme'] ) ? $request['theme'] : wp_get_theme()->get_stylesheet(),
-			);
 		} elseif ( 'custom' !== $template->source ) {
-			$changes->post_name   = $template->slug;
 			$changes->post_type   = $this->post_type;
 			$changes->post_status = 'publish';
-			$changes->tax_input   = array(
-				'wp_theme' => $template->theme,
-			);
```

Comment:
Seems like we can now merge these two branches into one:

```php
if ( null === $template || 'custom' !== $template->source ) {
```

Reply:
Amended.